include/m_field_info.h

```diff
@@ -7,10 +7,37 @@
 
 struct Game_Play;
 
+typedef enum fieldType {
+  /* 0 */ mFI_FIELDTYPE_FG,
+  /* 1 */ mFI_FIELDTYPE_1,
+  /* 2 */ mFI_FIELDTYPE_2,
+  /* 3 */ mFI_FIELDTYPE_ROOM,
+  /* 4 */ mFI_FIELDTYPE_NPC_ROOM,
+  /* 5 */ mFI_FIELDTYPE_DEMO,
+  /* 6 */ mFI_FIELDTYPE_PLAYER_ROOM,
+  /* 7 */ mFI_FIELDTYPE_NUM,
+}fieldType;
+
+
+#define mFI_TO_FIELD_ID(type, index) (((type) << 12) | (index))
+
+typedef enum fieldRoom{
+  /* 0x6000 */ mFI_FIELD_PLAYER0_ROOM = mFI_TO_FIELD_ID(mFI_FIELDTYPE_PLAYER_ROOM, 0),
+  /* 0x6001 */ mFI_FIELD_PLAYER1_ROOM,
+  /* 0x6002 */ mFI_FIELD_PLAYER2_ROOM,
+  /* 0x6003 */ mFI_FIELD_PLAYER3_ROOM,
+}fieldRoom;
```

Comment:
Actually, is this whole enum used anywhere besides the macro?
Feels like an extra layer of indirection that makes stuff a bit confusing

Reply:
It's used in every type of FieldType, but at the moment I only put the player types since that's what cockroach uses